provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/client/RestVlerClient.java

```diff
@@ -0,0 +1,172 @@
+    this.publicKey = publicKey;
+    this.privateKey = privateKey;
+    this.vlerTruststorePassword = vlerTruststorePassword;
+    this.vlerTruststore = vlerTruststore;
+    this.restTemplate = secureRestTemplate();
+  }
+
+  @SneakyThrows
+  private static <T extends VlerResponse> T handleVlerExceptions(
+      String message, Callable<T> callable) {
+    T response;
+    try {
+      response = callable.call();
+    } catch (HttpClientErrorException.NotFound e) {
+      throw new NotFound(message, e);
+    } catch (HttpClientErrorException.BadRequest e) {
+      throw new BadRequest(message, e);
+    } catch (HttpStatusCodeException e) {
+      throw new SearchFailed(message, e);
+    } catch (Exception e) {
+      throw new VlerException(message, e);
+    }
+    if (response == null) {
+      throw new VlerException(message + ", no VLER response");
+    }
+    return response;
+  }
+
+  @SneakyThrows
+  private String authHeader(String baseUrl, String url, String dateString) {
+    String endpoint;
+    try {
+      endpoint = "/" + url.substring(url.indexOf(baseUrl) + baseUrl.length());
+    } catch (Exception e) {
+      throw new BadRequest("Base URL not found within url ", e);
+    }
+    String reqStr = "GET\n" + dateString + "\napplication/json\n" + endpoint;
+    Mac encoding = Mac.getInstance("HmacSHA256");
+    encoding.init(new SecretKeySpec(privateKey.getBytes(Charset.forName("UTF-8")), "HmacSHA256"));
+    byte[] sha = encoding.doFinal(reqStr.getBytes(Charset.forName("UTF-8")));
+    String encSha = Base64.getEncoder().encodeToString(sha);
+    return "DAAS " + publicKey + ":" + encSha;
+  }
+
+  /** Calls the VLER Direct API by email address search. */
+  @Override
+  public AddressResponse endpointByAddress(String address) {
+    return handleVlerExceptions(
+        address,
+        () -> {
+          String url =
+              UriComponentsBuilder.fromHttpUrl(baseUrl + "/direct/addresses")
+                  .queryParam("cn", address)
+                  .build()
+                  .toUriString();
+          String dateString =
+              DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss z").format(ZonedDateTime.now());
+          HttpHeaders headers = new HttpHeaders();
+          headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+          headers.setContentType(MediaType.APPLICATION_JSON);
+          headers.add("Authorization", authHeader(baseUrl, url, dateString));
+          headers.add("Date", dateString);
+          ResponseEntity<AddressResponse> entity =
+              secureRestTemplate()
+                  .exchange(url, HttpMethod.GET, new HttpEntity<>(headers), AddressResponse.class);
+          return entity.getBody();
+        });
+  }
+
+  @SneakyThrows
+  private RestTemplate secureRestTemplate() {
+    ClassLoader cl = getClass().getClassLoader();
+    if (cl == null) {
+      throw new ClassLoaderException("Something went wrong getting the class loader");
```

Comment:
I know this from queen-elizabeth, but I think the dedicated `ClassLoaderException` class is overkill. Let's remove that and just throw a more generic `RuntimeException` instead.